### PR TITLE
Option to toggle invert scrolling of ASCII graph

### PR DESCRIPTION
**Pull request by Svenito**
Toggles both HJKL and mouse wheel scrolling of canvas

I'm not a fan of what Apple call "natural scrolling". I prefer scrolling motions to follow cursor movement. So I thought I would add an option to invert the default scrolling behaviour of the ASCII graph.

### CODE REVIEW

**Comment by ret2libc** [comment]
I think it's better to keep only one option to toggle between invert/normal. Otherwise it can happen that you set graph.invert_scroll, then with 'd' you change it and you end with an inconsistent view. Probably keep only the graph.invert_scroll is enough.

(btw, @radare i don't remember the rules for variable names. Isn't it better to avoid underscores?)

**Comment by Svenito** [comment]
Sounds good. Tbh you either want it or you don't, so removing the d toggle
makes sense.

I wasn't sure about the underscore either. I saw some other vars with
underscores in the file, so I thought it would be ok. Might rename to
'invscroll' though if ok?

File: libr/core/graph.c [hunk 1]
```diff
@@ -2036,6 +2036,7 @@ R_API int r_core_visual_graph(RCore *core, RAnalFunction *_fcn, int is_interacti
 		goto err_graph_new;
 	}
 	g->movspeed = r_config_get_i (core->config, "graph.scroll");
+	g->invscroll = r_config_get_i (core->config, "graph.invscroll");
```
Comment (ret2libc) [inline libr/core/graph.c]: there's no need to add invscroll to the RAGraph structure

Comment (ret2libc) [inline libr/core/graph.c]: Also, if you move the r_config_get_i inside the while loop you can get the change in the scroll direction without having to exit the ascii-art graph mode.

Comment (Svenito) [inline libr/core/graph.c]: Very true. Still getting my head around radare code, commits coming soon

On 16 July 2015 at 22:22, Riccardo Schirone <notifications@github.com>
wrote:

> In libr/core/graph.c
> <https://github.com/radare/radare2/pull/2962#discussion_r34839422>:
>
> > @@ -2036,6 +2036,7 @@ R_API int r_core_visual_graph(RCore *core, RAnalFunction *_fcn, int is_interacti
> >  		goto err_graph_new;
> >  	}
> >  	g->movspeed = r_config_get_i (core->config, "graph.scroll");
> > +	g->invscroll = r_config_get_i (core->config, "graph.invscroll");
>
> Also, if you move the r_config_get_i inside the while loop you can get the
> change in the scroll direction without having to exit the ascii-art graph
> mode.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/radare/radare2/pull/2962/files#r34839422>.
>



-- 
./Sven

Comment (ret2libc) [inline libr/core/graph.c]: No problem! Thanks a lot ;)

File: libr/core/graph.c [hunk 2]
```diff
@@ -2012,6 +2012,7 @@ R_API int r_core_visual_graph(RCore *core, RAnalFunction *_fcn, int is_interacti
 	int wheelspeed;
 	int w, h;
 	int ret;
+    int invscroll;
```
Comment (ret2libc) [inline libr/core/graph.c]: Really last thing :P wrong indentation here and

File: libr/core/graph.c [hunk 3]
```diff
@@ -2047,6 +2048,7 @@ R_API int r_core_visual_graph(RCore *core, RAnalFunction *_fcn, int is_interacti
 	core->cons->event_resize = (RConsEvent)agraph_refresh;
 
 	while (!exit_graph && !is_error) {
+        invscroll = r_config_get_i (core->config, "graph.invscroll");
```
Comment (ret2libc) [inline libr/core/graph.c]: here

**Comment by ret2libc** [comment]
Once indentation is fixed, it looks good to me!

**Comment by ret2libc** [comment]
Squashed and merged.
Thank you!

**Pull request closed without merging**